crates/pixi_build_frontend/src/tool/cache.rs

```diff
@@ -142,16 +297,56 @@ mod tests {
         ChannelConfig, MatchSpec, NamedChannelOrUrl, ParseStrictness, Platform,
     };
     use reqwest_middleware::ClientWithMiddleware;
-    use tokio::sync::{Barrier, Mutex};
+    use tokio::sync::{Barrier, Mutex, Semaphore};
 
     use crate::{
         tool::{
+            cache::{find_spec_records, ToolCache},
             installer::{ToolContext, ToolInstaller},
             IsolatedTool, ToolSpec,
         },
         IsolatedToolSpec,
     };
 
+    const BAT_META_JSON: &str = "bat-0.24.0-h3bba108_1.json";
+
+    /// A test helper to create a temporary directory and write conda meta files.
+    /// This is used to simulate already installed tools.
+    struct CondaMetaBuilder {
```

Comment:
Maybe something like `CondaMetaWriter` instead as I would expect with this name that it would actually construct the CondaMeta structs.